### PR TITLE
SA Checks: fromCallable

### PR DESCRIPTION
**Pull request by AlexandruGG**
Continuing the static analysis checks, this PR adds `fromCallable`.
- [x] adding check
- [x] improving tests
- [x] improving type hints

### CODE REVIEW

File: phpstan-baseline.neon [hunk 1]
```diff
@@ -5,33 +5,18 @@ parameters:
...
 			path: src/Collection.php
 
 		-
-			message: "#^Method loophp\\\\collection\\\\Collection\\:\\:getIterator\\(\\) return type has no value type specified in iterable type Iterator\\.$#"
-			count: 1
-			path: src/Collection.php
-
-		-
-			message: "#^Return type \\(Iterator\\) of method loophp\\\\collection\\\\Collection\\:\\:getIterator\\(\\) should be covariant with return type \\(Iterator\\<mixed, mixed\\>\\) of method loophp\\\\collection\\\\Contract\\\\Collection\\<TKey,T\\>\\:\\:getIterator\\(\\)$#"
+			message: "#^Parameter \\#1 \\$callable of class loophp\\\\collection\\\\Collection constructor expects callable\\(\\.\\.\\.mixed\\)\\: iterable\\<int, string\\>, Closure\\(string, string\\)\\: loophp\\\\collection\\\\Iterator\\\\StringIterator\\<mixed, string\\> given\\.$#"
```
Comment (AlexandruGG) [inline phpstan-baseline.neon:19]: This seems to be an issue with PHPStan, I've reported it here: https://github.com/phpstan/phpstan/issues/5060

**Review by drupol — APPROVED** [review]
Super great ! Thanks !

**Comment by AlexandruGG** [comment]
@drupol I just realised that in CI PHPStan doesn't run on the `tests/static-analysis` directory because it's ignored here: https://github.com/drupol/php-conventions/blob/master/config/php73/grumphp.yml#L85. I've been using it directly rather than via GrumPHP on my local so far and overlooked that.

How can we override that config for GrumPHP? Do we need to have a `tasks.phpstan.ignore_patterns` entry in the project's config where we list all paths again apart from the tests one?

**Comment by drupol** [comment]
Yes that's it.

**Pull request merged**